### ISSUE TITLE
plt.xticks after plotting with DateTimeIndex results in strange labels

### ISSUE THREAD

**Issue by toobaz** [pandas-dev/pandas#18881]
#### Code Sample, a copy-pastable example if possible

```python
N = 10
idx = pd.DatetimeIndex(['2015-{}-01'.format(i) for i in range(1, N+1)])
pd.Series(range(10), idx).plot()
```
results in the following:
[ex1](https://user-images.githubusercontent.com/1224492/34227782-b17faa7c-e5ce-11e7-93bc-122bfc719700.png)

In order to remove some labels on the x axis, I do
```python
plt.xticks(['2015-{}-01'.format(i) for i in range(1, N+1, 3)], fontsize=16)
```
... but the following happens:

![ex2](https://user-images.githubusercontent.com/1224492/34227838-df3bd1de-e5ce-11e7-8280-ed8d541a492d.png)

So I think "pandas formatter doesn't like to be customized, fine", and I add
```python
plt.gca().xaxis.set_major_formatter(DateFormatter('%Y-%m'))
```
and I get
![ex3](https://user-images.githubusercontent.com/1224492/34227876-04188e2a-e5cf-11e7-9623-d824143b709d.png)

If instead I directly plot with matplotlib:
```python
plt.plot(idx, range(10))
plt.xticks(['2015-{}-01'.format(i) for i in range(1, N+1, 3)], fontsize=16)
plt.gca().xaxis.set_major_formatter(DateFormatter('%Y-%m'))
```
I get the correct result:
![ex4](https://user-images.githubusercontent.com/1224492/34227960-4f0ee348-e5cf-11e7-8db7-6f6172fdb939.png)

#### Problem description

It should be possible to customize the xticks of the result of ``.plot()`` with the standard commands, at least when one accepts to completely discard the standard pandas formatter.

#### Expected Output

More ore less the last one

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: 775099cde4a2424dfb369d79b8b6e2687e5af7f9
python: 3.5.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.0-4-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: it_IT.UTF-8
LOCALE: it_IT.UTF-8

pandas: 0.22.0.dev0+393.g775099cde
pytest: 3.2.3
pip: 9.0.1
setuptools: 36.7.0
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.0dev
tables: 3.3.0
numexpr: 2.6.1
feather: 0.3.1
matplotlib: 2.0.0
openpyxl: 2.3.0
xlrd: 1.0.0
xlwt: 1.3.0
xlsxwriter: 0.9.6
lxml: 4.1.1
bs4: 4.5.3
html5lib: 0.999999999
sqlalchemy: 1.0.15
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: 0.2.1

</details>

**Comment by sinhrks** [comment]
This is because pandas formatter uses major and minor labels with different internal location from python datetime.

```
ax = pd.Series(range(10), idx).plot()
[str(x) for x in ax.get_xticklabels()]
# ["Text(540,0,'Jan\\n2015')"]
[str(x) for x in ax.get_xticklabels(minor=True)]
# ["Text(540,0,'')", "Text(541,0,'Feb')", "Text(542,0,'Mar')", "Text(543,0,'Apr')", "Text(544,0,'May')", "Text(545,0,'Jun')", "Text(546,0,'Jul')", "Text(547,0,'Aug')", "Text(548,0,'Sep')", "Text(549,0,'Oct')"]
````

You can do:

```
ax = pd.Series(range(10), idx).plot()
ax.set_xticks([539 + i for i in range(1, 10, 2)])
ax.set_xticklabels(['2015-{}'.format(i) for i in range(1, 10, 2)])
ax.minorticks_off()
```

**Comment by jorisvandenbossche** [comment]
Yes, this is indeed because pandas uses a combination of minor and major ticks for the formatting of the axis. Not sure if there is any action to be taken here? 
Some documentation about this (and how to customize) might be useful.

**Comment by 0x6d696368** [comment]
> Some documentation about this (and how to customize) might be useful.

Yes definitively! I'm scratching my head at how I can make the following work as intended:
```
import matplotlib.pyplot as plt
import pandas as pd
from io import StringIO

df = pd.read_csv(StringIO("""time;a
    2020-02-20 20:20:20;foo
    2020-02-20 21:20:20;bar
    2020-02-20 21:20:20;foo
    2020-02-20 22:20:20;foo
    2020-02-20 23:20:20;bar
    """),sep=';')
df['time'] = pd.to_datetime(df['time'], format="%Y-%m-%d %H:%M:%S.")
df2 = df.groupby(['time','a'])['a'].count().mul(5).unstack('a').fillna(0)
df3 = df2.resample('30T').sum()

ax = df3.plot(kind='area',stacked=True)
# How can I get a result equal to the following two lines?
#ax.xaxis.set_major_locator(mdates.AutoDateLocator())
#ax.xaxis.set_major_formatter(mdates.DateFormatter('%Y-%m-%d %H:%M'))

plt.show()
```
Any recommendation would be appreciated.
